index.js

```diff
@@ -212,6 +212,17 @@
     function(x) { return x != null; }
   );
 
+  //  Applicative :: TypeClass
+  var Applicative = $.TypeClass(
+    'sanctuary/Applicative',
+    function(x) {
+      return (
+        _type(x) === 'Array' ||
+        Apply.test(x) && (hasMethod('of')(x) || hasMethod('of')(x.constructor))
+      );
+    }
+  );
+
```

Comment:
I prefer the second approach too. It would make it possible to create a graph of the relationships, as dependencies could be exposed via a field on a TypeClass value rather than being hidden in its predicate.